documentation/freshdesk.md

```diff
@@ -0,0 +1,34 @@
+# Freshdesk
+
+Freshdesk is the customer service software used by GC Digital Talent for creating support tickets.
+
+## Account
+
+1. Navigate to [Freshdesk](https://www.freshworks.com/freshdesk/signup/)
+2. Sign up for an account
+3. Make note of the _domain_ upon account creation
+4. Create custom string field for the ticket object named `page_url` (https://_domain_.freshdesk.com/a/admin/ticket_fields)
+
+## Local setup
+
+### Environment variables
+
+These values are read by the API in `api/config/freshdesk.php`. The API logic for creating a ticket can be found within `api/app/Http/Controllers/SupportController.php` in the `createTicket` method. The frontend logic for creating a ticket can be found within `apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx`.
+
+#### `api/.env`
+
+- Set value of `FRESHDESK_API_TICKETS_ENDPOINT` to "https://_domain_.freshdesk.com/api/v2/tickets" where the _domain_ value can be found in [Freshdesk account](https://support.freshdesk.com/en/support/solutions/articles/237264-how-do-i-find-my-freshdesk-account-url-using-my-email-address-)
+- Set value of `FRESHDESK_API_KEY` to string found in [Freshdesk account](https://support.freshdesk.com/en/support/solutions/articles/215517-how-to-find-your-api-key)
+- Set value of `FRESHDESK_API_TICKET_TAG` to "test"
+
+> [!NOTE]  
+> Optional fields that are not required for ticket creation locally but are used in production are `FRESHDESK_API_PRODUCT_ID` and `FRESHDESK_API_EMAIL_CONFIG_ID`. The Freshdesk API has endpoints for [products](https://developers.freshdesk.com/api/#products) and [email configs](https://developers.freshdesk.com/api/#email-configs) in case they need to be tested on a local environment
```

Comment:
Totally unrelated to this PR, but did you see in the docs that email-configs is deprecated?  I wonder if we'll need to update our controller at some point.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/2b498d16-78ea-4dee-801a-35df31b08619)

Reply:
I did see that in the docs, awhile ago actually. I've held off doing anything since it was still serving some useful function the last time I tried to remove it (Freshdesk's API documentation leaves a little bit to be desired). If you'd like to create an issue to try and replace the email config with an email mailbox, go for it, though it will require some access to the production account's dashboard for testing and configuration.

Reply:
Sounds good.  I opened https://github.com/GCTC-NTGC/gc-digital-talent/issues/10153 though it's probably not very high priority.